lntest/itest/lnd_misc_test.go

```diff
@@ -1308,6 +1308,180 @@ func testRejectHTLC(net *lntest.NetworkHarness, t *harnessTest) {
+	const chanAmt = btcutil.Amount(1000000)
+	ctxb := context.Background()
+
+	carol := net.NewNode(t.t, "Carol", nil)
+	defer shutdownAndAssert(net, t, carol)
+
+	// Connect Carol to Bob.
+	net.ConnectNodes(ctxb, t.t, net.Bob, carol)
+
+	// Send coins to Alice.
+	net.SendCoins(ctxb, t.t, btcutil.SatoshiPerBitcoin, net.Alice)
+
+	// Send coins to Bob.
+	net.SendCoins(ctxb, t.t, btcutil.SatoshiPerBitcent, net.Bob)
+
+	// Open a channel between Alice and Bob.
+	ctxt, _ := context.WithTimeout(ctxb, channelOpenTimeout)
+	chanPointAlice := openChannelAndAssert(
+		ctxt, t, net, net.Alice, net.Bob,
+		lntest.OpenChannelParams{
+			Amt: chanAmt,
+		},
+	)
+
+	// Open a channel between Bob and Carol.
+	ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
+	chanPointCarol := openChannelAndAssert(
+		ctxt, t, net, net.Bob, carol,
+		lntest.OpenChannelParams{
+			Amt: chanAmt,
+		},
+	)
+
+	// Channel should be ready for payments.
+	const payAmt = 2500
+
+	// Helper closure to generate a random pre image.
+	genPreImage := func() []byte {
+		preimage := make([]byte, 32)
+
+		_, err := rand.Read(preimage)
+		if err != nil {
+			t.Fatalf("unable to generate preimage: %v", err)
+		}
+
+		return preimage
+	}
+
+	// Once we disable the channel, we won't be able to simply ask the node
+	// to pay Carol's invoice since disabling a channel updates other nodes
+	// in the network. To get around this, we'll construct our path before
+	// the channel is disabled.
+	sendMPP := func(sender, reciever *lntest.HarnessNode, payAmt int64) bool {
+		preimage := genPreImage()
+
+		// Create an invoice from Carol and add it to her database
+		recieverInvoice := &lnrpc.Invoice{
+			Memo:      "testing - bob should pay carol",
+			RPreimage: preimage,
+			Value:     payAmt,
+		}
+		resp, err := reciever.AddInvoice(ctxb, recieverInvoice)
+		if err != nil {
+			t.Fatalf("unable to add invoice: %v", err)
+		}
+
+		// Construct a route from Alice to Carol
+		routesReq := &lnrpc.QueryRoutesRequest{
+			PubKey:         reciever.PubKeyStr,
+			Amt:            payAmt,
+			FinalCltvDelta: 144,
+		}
+		ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+		routes, err := sender.QueryRoutes(ctxt, routesReq)
+
+		if err != nil {
+			t.Fatalf("unable to get route: %v", err)
+		}
+
+		if len(routes.Routes) != 1 {
+			t.Fatalf("expected to find 1 route, got %v", len(routes.Routes))
+		}
+
+		// Multi Path Payments are default now, so we're marking the last hop
+		// with an MPP Record containing carol's address
+		hop := routes.Routes[0].Hops[len(routes.Routes[0].Hops)-1]
+		hop.TlvPayload = true
+		hop.MppRecord = &lnrpc.MPPRecord{
+			PaymentAddr:  resp.PaymentAddr,
+			TotalAmtMsat: payAmt * 1000,
+		}
+
+		// There should be only one valid route, so we'll use the first element
+		sendReq := &routerrpc.SendToRouteRequest{
+			PaymentHash: resp.RHash,
+			Route:       routes.Routes[0],
+		}
+		_, err = sender.RouterClient.SendToRouteV2(ctxb, sendReq)
+		if err != nil {
```

Comment:
We started using the `require` library for new code. It's quite nice as it shortens the code quite a bit and also produces nicer error messages.